runtime/moonbeam/tests/integration_test.rs

```diff
@@ -1124,7 +1124,11 @@ fn ethereum_invalid_transaction() {
 		// Ensure an extrinsic not containing enough gas limit to store the transaction
 		// on chain is rejected.
 		assert_eq!(
-			Executive::apply_extrinsic(unchecked_eth_tx(INVALID_ETH_TX)),
+			Executive::validate_transaction(
```

Comment:
Is not possible to test using `apply_extrinsic` anymore?

Reply:
No, because this check is not done by apply_extrinsic anymore.
There are two different contexts: the pre-validation context (done by the pool), and the execution context. Before, we used to call validate_unsigned in both contexts, now it is only in the context of validation by the pool.

Reply:
Not sure I understand, how the relay would be aware of invalid transactions if the custom validation logic is not called on `apply_extrinsic`?

Reply:
In this case, the transaction will still be rejected at execution by the evm, with the BalanceLow error.
The purpose of the validation pre-check is to help the creator of a block to check if a transaction is valid without executing it, which is necessary to avoid DOSing the pool with lots of invalid transactions.
But once the transaction is in the block, these pre-checks are no longer useful, they increase the execution time of the block for nothing.

Reply:
But maybe some of the checks performed by `validate_self_contained` should still be performed in `apply_self_contained` for security reasons, @nanocryk what is your opinion on this?

Reply:
Can you add a test to prove there are no unwanted side-effects with the new approach?

For example a test that still uses `apply_extrinsic` and proves there's no side-effect at pallet-ethereum storage level, like stored transactions that couldn't afford storing the transaction on chain in the first place (not enough provided gas limit to pay the transaction cost).